packages/web-lib/hooks/useChain.tsx

```diff
@@ -0,0 +1,23 @@
+import {chains, TChain} from '@yearn-finance/web-lib/utils/web3/chains';
+import { TNDict } from '../utils/types';
+import { useChainID } from './useChainID';
+
+type TUseChainReturn = {
+	get: (chainID: number) => TChain;
+	getCurrent: () => TChain;
+	getAll: () => TNDict<TChain>;
+}
+
+/* 🔵 - Yearn Finance ******************************************************
+** This hook can be used to grab details about a network.
+** It will return details of the network.
+**************************************************************************/
+export function useChain(): TUseChainReturn {
+	const {safeChainID} = useChainID();
+
+	return {
+		get: (chainID: number): TChain => chains[chainID],
+		getCurrent: (): TChain => chains[safeChainID],
+		getAll: () => chains
```

Comment:
@Majorfi I went with this approach instead because we cannot call hooks inside conditions, and there are a few places where we either want all the chain details or we want to get details about a different chain. Wdyt?